src/botmsg.c

```diff
@@ -755,7 +755,7 @@ int add_note(char *to, char *from, char *msg, int idx, int echo)
 {
   #define FROMLEN 40
   int status, i, iaway, sock;
-  char *p, botf[FROMLEN + 1 + HANDLEN + 1], ss[81], ssf[20 + 1 + sizeof botf];
+  char *p, botf[FROMLEN + 1 + HANDLEN + 1], ss[21], ssf[20 + 1 + sizeof botf], *endptr;
```

Comment:
why is 21 correct?

Reply:
ss is used as the destination of long-> string conversion. 21 is exactly that. try to lower it by one and your compiler will likely yell at you. longer numbers will be truncated there which is fine.